lib/oracle-store.js

```diff
@@ -1,352 +1,420 @@
-  desc = meta.desc
+  var meta = seneca.store.init(seneca,opts,store);
+  desc = meta.desc;
 
 
   seneca.add({init:store.name,tag:meta.tag},function(args,done){
     configure(opts,function(err){
-      if( err ) return seneca.die('store',err,{store:store.name,desc:desc});
+      if( err ) {
+          return seneca.die('store',err,{store:store.name,desc:desc});
+      }
       return done();
-    })
-  })
-
-
-  return {name:store.name,tag:meta.tag}
-}
+    });
+  });
+
+
+  return {name:store.name,tag:meta.tag};
+};
+
+/**
+ * Maps an OracleDB results to an array of rows (name: value);
+ * The OracleDB result is in the form:
+ * { rows: [ [ '41f47bfa-fcde-4c89-8a1b-7907e33469fb', 'v1', 'v1', null ] ],
+ * resultSet: undefined,
+ * outBinds: undefined,
+ * rowsAffected: undefined,
+ * metaData:
+ *  [ { name: 'id' },
+ *   { name: 'p1' },
+ *   { name: 'p2' },
+ *   { name: 'seneca' } ] }
+ */
+var mapOracleResultToRow = function (results) {
+
+    // Map the metadata to a row (name=value).
+    var columns = results.metaData.map(function (item) {
+        return item.name;
+    });
+
+    var rows = results.rows.map(function (row) {
+        return columns.reduce(function (prev, column, index) {
+            prev[column] = row[index];
+            return prev;
+        }, {});
+    });
+
+    return rows;
+};
 
 var makeentp = function(ent) {
-  var entp   = {}
-  var type   = {}
-  var fields = ent.fields$()
+  var entp   = {};
+  var type   = {};
+  var fields = ent.fields$();
 
-  fields.forEach(function(field){
+  fields.forEach(function(field) {
     if( _.isDate( ent[field ]) ) {
       type[field] = DATE_TYPE;
-      entp[field] = ent[field]
+      entp[field] = ent[field];
     }
     else if( _.isArray( ent[field] ) ) {
       type[field] = ARRAY_TYPE;
-      entp[field] = JSON.stringify(ent[field])
+      entp[field] = JSON.stringify(ent[field]);
     }
     else if( _.isObject( ent[field] ) ) {
       type[field] = OBJECT_TYPE;
-      entp[field] = JSON.stringify(ent[field])
+      entp[field] = JSON.stringify(ent[field]);
     }
     else if( _.isBoolean( ent[field] ) ) {
       type[field] = BOOLEAN_TYPE;
-      entp[field] = ent[field] ? 'T' : 'F'
+      entp[field] = ent[field] ? 'T' : 'F';
     }
     else if( _.isUndefined( ent[field] ) ) {
-      entp[field] = null
+      entp[field] = null;
     }
     else {
-      entp[field] = ent[field]
+      entp[field] = ent[field];
     }
-  })
+  });
 
   if ( !_.isEmpty(type) ){
-    entp[SENECA_TYPE_COLUMN] = JSON.stringify(type)
+    entp[SENECA_TYPE_COLUMN] = JSON.stringify(type);
   }
 
-  return entp
-}
+  return entp;
+};
+
+/**
+ * Map an array of rows to an arrays of entities, using makeent.
+ */
+var makeentRows = function(ent, rows, callback) {
 
-var makeent = function(ent,row) {
-  var entp       = {}
-  var senecatype = {}
-  var fields      = _.keys(row)
+    async.map(rows, function (row, cb) {
+        makeent(ent, row, function (err, res) {
```

Comment:
why not just calling `makeent(ent, row, cb);`? 
We could even do `async.map(rows, async.apply(makeent, ent));`.